src/include/sof/coherent.h

```diff
@@ -158,16 +171,12 @@ static inline struct coherent *coherent_release_irq(struct coherent *c, const si
 		object->member.shared = false;				\
 		object->member.core = cpu_get_id();				\
 		list_init(&object->member.list);				\
-		/* wtb and inv local data to coherent object */			\
-		dcache_writeback_invalidate_region(uncache_to_cache(object), sizeof(*object)); \
 	} while (0)
 
 #define coherent_free(object, member)			\
 	do { \
 		/* assert if someone passes a cache address in here. */	\
 		ADDR_IS_COHERENT(object);					\
-		/* wtb and inv local data to coherent object */			\
-		dcache_writeback_invalidate_region(uncache_to_cache(object), sizeof(*object)); \
```

Comment:
Needed here.

Reply:
@lgirdwood why? this is a `free()`. We finished using the memory already and released it. When we released it all caches have been synchronised and dropped. We cannot free such memory while it is used / not released.

Reply:
The next user may not use the coherent API.

Reply:
@lgirdwood sorry, I was trying to say, that we have no cached data at this point, there is nothing to write back or invalidate. This function - `coherent_free()` - can only be called either if we didn't use the memory at all, immediately after `coherent_init()` or after we used it and so we called `coherent_acquire()` / `coherent_release()` for every such use. The last  call to `coherent_release()` has already invalidated caches. We have no cached data of this memory, nothing to write back or invalidate.
In fact, the next user either must continue using the coherent API - as long as this memory is kept, or this memory must be freed back to the heap, at which point caches will be invalidated again.